### PR TITLE
leverage external events channel

### PR DESCRIPTION
**Pull request by compleatang**
Signed-off-by: Casey Kuhlman <casey@monax.io>

### CODE REVIEW

**Comment by silasdavis** [comment]
I think this looks reasonable. The idea is that we are able to intercept vent events post-persist post-decode. Vent takes care of interpreting the spec and remembering the block height offset it is at. It also generates a database mapping - which may be seen as side-effect if all you really want is the events.

 My only possible query is that now we essentially have two integration points into vent - the `DBAdapter` and this `EventChannel`. Part of me wonders if it would be cleaner to provide both functionalities by allowing Vent to write to multiple `DBAdaptors`. Having said that implementing a full `DBAdaptor` for this feels like overkill so perhaps this is a perfectly clean way of integrating there.

So basically LGTM but would like @hiturria and @seanyoung's thoughts..

**Comment by compleatang** [comment]
I think it would be a relatively rare occurrence where the user that turned on the `externalChannel` would not want *something* of a structured DB locally. 

For that reason, I'm not sure I see the benefit of a full `DBAdapter` for simply sending an event to a system that was leveraging vent as a library, nor can I think of a time in which it would be advantageous for vent to have to write to multiple DBs within the same vent instance (in such a scenario it would be much more resilient to run multiple vents writing to different DBs). 

Not necessarily saying the points were off base, just my comments.

**Comment by silasdavis** [comment]
Yeah I agree, just wanted to push on that to make sure we weren't adding 'many ways to do the same thing'. Satisfied this is not a case of this. I think it's going to be a useful feature.
